### PR TITLE
pkg: remove not needed git-downloaded dependencies

### PR DESCRIPTION
**Pull request by kaspar030**
# Contribution description

Currently, "git-dowloaded" is evaluated twice, one time by the "pkg-prepare"-step of the main make
instance, then again in the build step of the package.

Since "git-ensure-version" this makes all packages that have either patches or a tag as version get checked out twice.

Anyhow, packages are not supposed to be built outside the build system, so doing the check twice doesn't make sence. Thus this PR removes the dependency on all the package's "all:" target.

### Testing procedure

Let CI build all packages.

### Issues/PRs references

#11129

### CODE REVIEW

**Comment by cladmi** [comment]
Agreed. These are useless. The preparation must always have been done as part of the `BUILDDEPS/APPDEPS` as other files depends on the header being available.

Please remove the `pkg/wac` that got included here. You can inline remove it.
(I cannot ACK until then).

----- 
Side note.
What would be useful, would be that a created `file` target depends on a `.git-prepared` file target to  allow not rebuilding. But we are really far away.

**Comment by kaspar030** [comment]
> Please remove the `pkg/wac` that got included here. You can inline remove it.

Well spotted, thanks. Removed it.

File: pkg/emb6/Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ EMB6_SUBMODULES:=$(filter-out emb6_contrib \
 
 .PHONY: all
 
-all: git-download $(EMB6_SUBMODULES)
+all: $(EMB6_SUBMODULES)
 	"$(MAKE)" -C $(PKG_BUILDDIR)
 
 # Rule for all submodules
```
Comment (cladmi) [inline pkg/emb6/Makefile]: Should also remove the `emb6_%: git-download` here below.

**Comment by cladmi** [comment]
To note, there are still other usages

```
git grep 'git-download' '*' ':!pkg/local.mk' ':!pkg/pkg.mk'
dist/tools/bossa/Makefile:all: git-download
dist/tools/edbg/Makefile:all: git-download
dist/tools/mosquitto_rsmb/Makefile:all: git-download
dist/tools/teensy-loader-cli/Makefile:all: git-download
pkg/ccn-lite/Makefile:$(TOOLCHAIN_FILE): git-download
pkg/ccn-lite/Makefile:git-download: | ..cmake_version_supported
pkg/emb6/Makefile:emb6_%: git-download
pkg/relic/Makefile:$(TOOLCHAIN_FILE): git-download
pkg/relic/Makefile:git-download: | ..cmake_version_supported
pkg/wakaama/Makefile:patch: git-download
```

* pkg/emb6 should be updated

Do not need to be updated:

* `dist/tools` they require depending on `git-download` as there is no `prepare` step.

Not direct migrations cases that should go in dedicated PRs:

* `pkg/ccn-lite` and `pkg/relic` could remove the usage. But it means that `$(TOOLCHAIN_FILE)` should depend on `FORCE` to have the same behavior and that `..cmake_version_supported` must be done before `prepare`.
* pkg/wakaama: `patch` must indeed depend on `git-download`. But would need to be  a target on `prepare` instead, so different fix than here

**Comment by kaspar030** [comment]
>     * pkg/emb6 should be updated

fixed



> Not direct migrations cases that should go in dedicated PRs:

postponed

> Do not need to be updated:

agreed

**Comment by kaspar030** [comment]
- squashed

@cladmi you ok here?

**Pull request merged**